### PR TITLE
[ios] New way to editing names.

### PR DESCRIPTION
**Pull request by VladiMihaylenko**

### CODE REVIEW

File: iphone/Maps/Classes/Editor/MWMEditorViewController.mm [hunk 1]
```diff
@@ -423,14 +410,11 @@ - (void)configTable
 
   if (isNameEditable)
   {
-    m_sections.push_back(MWMEditorSectionName);
-    m_cells[MWMEditorSectionName] = kSectionNameCellTypes;
-    registerCellsForTableView(kSectionNameCellTypes, self.tableView);
-
-    vector<osm::LocalizedName> localizedNames = getAdditionalLocalizedNames(m_mapObject);
+    auto const ds = m_mapObject.GetNamesDataSource();
+    vector<osm::LocalizedName> localizedNames = ds.names;
```
Comment (milchakov) [inline iphone/Maps/Classes/Editor/MWMEditorViewController.mm]: auto const &

File: iphone/Maps/Classes/Editor/MWMEditorViewController.mm [hunk 2]
```diff
@@ -564,23 +548,12 @@ - (void)fillCell:(UITableViewCell * _Nonnull)cell atIndexPath:(NSIndexPath * _No
...
-               capitalization:UITextAutocapitalizationTypeSentences];
-    break;
-  }
   case MWMPlacePageCellTypeAdditionalName:
   {
     MWMEditorAdditionalNameTableViewCell * tCell =
         static_cast<MWMEditorAdditionalNameTableViewCell *>(cell);
 
-    vector<osm::LocalizedName> const localizedNames = getAdditionalLocalizedNames(m_mapObject);
+    vector<osm::LocalizedName> const localizedNames = m_mapObject.GetNamesDataSource().names;
```
Comment (milchakov) [inline iphone/Maps/Classes/Editor/MWMEditorViewController.mm]: auto

**Comment by milchakov** [comment]
LGTM

**Comment by igrechuhin** [comment]
LGTM

**Pull request merged**